Formula/coredns.rb

```diff
@@ -0,0 +1,89 @@
+
+class Coredns < Formula
+  desc "DNS server that chains plugins"
+  homepage "https://coredns.io"
+  url "https://github.com/coredns/coredns/archive/v1.0.3.tar.gz"
+  sha256 "dedc42f1a4aac56e397a3fff19955327f21460782aef870658fa1353134ab2f7"
+  head "https://github.com/coredns/coredns.git"
+
+  def default_coredns_config; <<~EOS
+    . {
+      hosts {
+        fallthrough
+      }
+      proxy . 8.8.8.8:53 8.8.4.4:53 {
+        protocol https_google
+      }
+      cache
+      errors
+    }
+    EOS
+  end
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    ENV["GOOS"] = "darwin"
+    ENV["GOARCH"] = MacOS.prefer_64_bit? ? "amd64" : "386"
+    ENV.prepend_path "PATH", buildpath/"bin"
+
+    (buildpath/"src/github.com/coredns/coredns").install buildpath.children
+
+    cd "src/github.com/coredns/coredns" do
+      # Replace git call with version
+      inreplace "Makefile", "$(shell git describe --dirty --always)", "v1.0.3"
+
+      system "make", "coredns"
+      sbin.install "coredns"
```

Comment:
You'll likely need to add a `prefix.install_metafiles` line here.

This ensures things like `README`, `LICENSE` and `CHANGELOG` where available are installed into the prefix. Normally Homebrew handles this automatically with a little magic, but when you relocate the entire contents of the `buildpath` to another directory for the install process as is common with `go`-using formulae that logic falls on its face & it needs a little manual nudging. It's a known issue.